src/Optimizer.ts

```diff
@@ -19,6 +19,12 @@ export class Optimizer {
   removeComponentsReport: ReportElement[] = [];
   moveToComponentsReport: ReportElement[] = [];
   outputObject = {};
+
+  /**
+   * You can initialize the Optimizer class using this constructor.
+   * @param {any} YAMLorJSON - YAML or JSON document that you want to optimize. you can pass Object, YAML or JSON version of your AsyncAPI document here.
+   */
+
```

Comment:
In JS you have only one constructor, not like in C++ multiple, so description is not needed here :)

```suggestion
  /**
   * @param {any} YAMLorJSON - YAML or JSON document that you want to optimize. You can pass Object, YAML or JSON version of your AsyncAPI document here.
   */
```